lightning/src/offers/invoice_request.rs

```diff
@@ -1537,6 +1538,23 @@ mod tests {
 		assert_eq!(tlv_stream.payer_note, Some(&String::from("baz")));
 	}
 
+	#[test]
+	fn compute_tagged_hash() {
+		let unsigned_invoice_request = OfferBuilder::new("foo".into(), recipient_pubkey())
+			.amount_msats(1000)
+			.build().unwrap()
+			.request_invoice(vec![1; 32], payer_pubkey()).unwrap()
+			.payer_note("bar".into())
+			.build().unwrap();
+
+		// Simply test that we can grab the tag and merkle root exposed by the accessor
+		// functions, then use them tosuccesfully compute a tagged hash.
+		let taggedhash = unsigned_invoice_request.as_ref();
+		let tag = sha256::Hash::hash(taggedhash.tag().as_bytes());
+		let _ = Message::from_slice(&tagged_hash(tag, taggedhash.merkle_root_hash()))
+			.unwrap();
```

Comment:
Not sure if I see much value in this test. How would it fail?

Reply:
Fair enough, my thought here was that this is how I'll roughly be using the new TaggedHash getters in LNDK. Tests that new() sets the right data and the new getters are retrieving the correct data and passing it into tagged_hash. But happy to remove if you prefer.

Reply:
Hmm... it essentially duplicates the body of `TaggedHash`/`message_digest` but doesn't test against the `Message` digest produced there to see if they are equal. If you add that assertion, you can move this test to the `merkle` module as a sanity test, I suppose. That way we don't need to expose any private functions.

Reply:
Sounds good, sanity test is a good phrase for what I was trying to do :)